src/backend/core/api/serializers.py

```diff
@@ -218,16 +219,36 @@ def validate_file(self, file):
                 f"File size exceeds the maximum limit of {max_size:d} MB."
             )
 
-        # Validate file type
-        mime_type, _ = mimetypes.guess_type(file.name)
-        if mime_type not in settings.DOCUMENT_IMAGE_ALLOWED_MIME_TYPES:
-            mime_types = ", ".join(settings.DOCUMENT_IMAGE_ALLOWED_MIME_TYPES)
-            raise serializers.ValidationError(
-                f"File type '{mime_type:s}' is not allowed. Allowed types are: {mime_types:s}"
-            )
+        extension = file.name.rpartition(".")[-1] if "." in file.name else None
+
+        # Read the first few bytes to determine the MIME type accurately
+        mime = magic.Magic(mime=True)
+        magic_mime_type = mime.from_buffer(file.read(1024))
+        file.seek(0)  # Reset file pointer to the beginning after reading
+
+        if magic_mime_type in settings.DOCUMENT_UNSAFE_MIME_TYPES:
+            self.context["is_unsafe"] = True
```

Comment:
Why don't you want to fill this information always? Like 
```suggestion
        self.context["is_unsafe"] = magic_mime_type in settings.DOCUMENT_UNSAFE_MIME_TYPES
```

As a slight optimization, the `settings.DOCUMENT_UNSAFE_MIME_TYPES` could be a `set` instead of a list.

Reply:
I suppose because semantically it's overstated to say they are "safe" just because we are not sure that they are "unsafe" ^^ But yes good idea.